### PR TITLE
Update for SourceKitten File lines

### PR DESCRIPTION
**Pull request by keith**
This depends on https://github.com/jpsim/SourceKitten/pull/62

This updates all of SwiftLint's calls to the now removed `lines()` function. Instead we're using the immutable cached property on File. This way we don't have to compute these as many times.

Right now this PR is pointing to my fork and the sha of that branch, I'll remove this change once that other PR is figured out.

### CODE REVIEW

File: Source/SwiftLintFramework/Rules/FileLengthRule.swift [hunk 1]
```diff
@@ -19,7 +19,7 @@ public struct FileLengthRule: ParameterizedRule {
     ]
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let lines = file.contents.lines()
+        let lines = file.lines
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: this can just be `let lineCount = file.lines.count`

Comment (keith) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: Nice, updating

**Comment by keith** [comment]
I've updated this to use the correct version of SourceKitten again, although I'm not sure how you want to handle that since it's pointing to master and not a specific release right now.

**Comment by jpsim** [comment]
The Cartfile should just point to `github jpsim/SourceKitten, "master"` until a release is made.

**Comment by keith** [comment]
I've updated the Cartfile.resolved to match master there, since master is already in the Cartfile. I saw some weirdness about the result library locally so lets see if this build passes.

**Comment by keith** [comment]
Looks like I hit this carthage issue https://github.com/Carthage/Carthage/issues/135 locally because I changed the submodule URL locally, I deleted that commit and everything looks good on CI though.

File: Source/SwiftLintFramework/Rules/FileLengthRule.swift [hunk 2]
```diff
@@ -19,14 +19,14 @@ public struct FileLengthRule: ParameterizedRule {
     ]
 
     public func validateFile(file: File) -> [StyleViolation] {
-        let lines = file.contents.lines()
+        let lines = file.lines.count
```
Comment (jpsim) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: please this variable to `lineCount` and we're in business!

**Comment by jpsim** [comment]
Other than `lineCount`, :+1:

File: Source/SwiftLintFramework/Rules/FileLengthRule.swift (hunk 2 above)
Comment (keith) [inline Source/SwiftLintFramework/Rules/FileLengthRule.swift]: Haha, I was just thinking this name was no longer obvious. Updated.

**Pull request merged**